### DISCUSSION TITLE
Why does the `setCookie` example divide `maxAge` by 1000?

### DISCUSSION THREAD

**Discussion by will-stone** [vercel/next.js#27617] [Help]
Hi 👋 

The example code for a `setCookie` function (https://nextjs.org/docs/api-routes/api-middlewares#extending-the-reqres-objects-with-typescript) shows the `maxAge`option being divided by 1000. Why is this done? Max-age is already in seconds, so I presume this is not for millisecond conversion...? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes

Thanks,

Will.

**Comment by will-stone** [comment]
I believe the function should be:

```js
if (typeof options.maxAge === 'number') {
  options.expires = new Date(Date.now() + options.maxAge * 1000);
}
```

The max-age attribute for Set-Cookie is in seconds. Therefore I believe there's a few changes that are required to the middle conditional:

1. The check should only be concerned with whether the `maxAge` is numerical.
2. As `Date.now` provides time in milliseconds, we need to multiply `options.maxAge` by 1000.
3. `options.maxAge` does not require altering as it can be passed on in seconds to the `serialize` function which expects it in seconds:
  - https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes
  - https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/cookie/index.d.ts#L55

**Comment by zmarty** [comment]
Opened related bug: https://github.com/vercel/next.js/issues/33472
